src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-edit-workingtime-dialog/exam-edit-working-time-button.directive.ts

```diff
@@ -0,0 +1,78 @@
+import { Directive, ElementRef, EventEmitter, HostListener, Input, OnDestroy, OnInit, Output, Renderer2 } from '@angular/core';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { from } from 'rxjs';
+
+import { ButtonSize, ButtonType } from 'app/shared/components/button.component';
+import { Exam } from 'app/entities/exam.model';
+import { ExamEditWorkingTimeDialogComponent } from 'app/exam/manage/exams/exam-checklist-component/exam-edit-workingtime-dialog/exam-edit-working-time-dialog.component';
+import { AlertService } from 'app/core/util/alert.service';
+import dayjs from 'dayjs/esm';
+
+@Directive({ selector: '[jhiEditWorkingTimeButton]' })
+export class ExamEditWorkingTimeButtonDirective implements OnInit, OnDestroy {
+    @Input() exam: Exam;
+    @Output() examChange = new EventEmitter<Exam>();
+
+    @Input() buttonSize: ButtonSize = ButtonSize.MEDIUM;
+    @Input() buttonType: ButtonType = ButtonType.WARNING;
+
+    private modalRef: NgbModalRef | null;
+
+    private intervalRef: any;
+
+    constructor(
+        private modalService: NgbModal,
+        public alertService: AlertService,
+        private renderer: Renderer2,
+        private elementRef: ElementRef,
+    ) {}
+
+    /**
+     * This method appends classes and type property to the button on which directive was used, additionally adds a span tag with delete text.
+     * We can't use component, as Angular would wrap it in its own tag and this will break button grouping that we are using for other buttons.
+     */
+    ngOnInit() {
+        this.renderer.addClass(this.elementRef.nativeElement, 'btn');
+        this.renderer.addClass(this.elementRef.nativeElement, this.buttonType);
+        this.renderer.addClass(this.elementRef.nativeElement, this.buttonSize);
+        this.renderer.setProperty(this.elementRef.nativeElement, 'type', 'submit');
+
+        this.checkWorkingTimeChangeAllowed();
+        this.intervalRef = setInterval(this.checkWorkingTimeChangeAllowed.bind(this), 1000);
+    }
```

Comment:
I'm confused by this and this directive in general. 

- There are other occasions in the application with the exact same comment above `ngOnInit` in the codebase. So I guess you copied it over, but this does not make sense here, does it? What "button grouping"? You only show a single button in a row of the exam checklist.
- Why do we have button size and -type inputs when we only use this button in a single spot?
- We should avoiding playing with `Renderer2` and `nativeElement` stuff unless required; everything that is done here can be added to the button declaration itself
- Let's convert this to a component instead of a directive. Just put the button into the template, and then add the component to the exam checklist. I don't see a reason for this to be a directive, and it might cause more confusion than a simple component as that's what we usually do

Reply:
Ok, totally agree. Yes, I've copied it from the delete dialog, and was not accurate enough with cleaning up. Also, I still have to get a feeling for what should be a component and what should be a directive. As it's just used once, one can also debate to put all of the code into the checklist component, but yea, I did not want to make it even larger.

Reply:
Incorporated your feedback. Made the directive a component. Removed all unnecessary code.